dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store.cpp

```diff
@@ -501,6 +499,99 @@ try
 }
 CATCH
 
+TEST_P(DeltaMergeStoreRWTest, ReadAfterLogicalSplit)
```

Comment:
Add a test to cover this case.
Without the fix in Segment.cpp, this case will report error as below
```
/DATA/disk1/jaysonhuang/tiflash/dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store.cpp:590: Failure

  (in).read() return more rows(49152) than expected
  (createColumns({ createColumn<Int64>(createNumbers<Int64>(0, num_rows_write)), })).rows()
    Which is: 24576
```